src/connector/src/sink/opensearch.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use futures::prelude::TryFuture;
+use futures::FutureExt;
+use opensearch::{BulkOperation, BulkParts, OpenSearch};
+use risingwave_common::array::StreamChunk;
+use risingwave_common::catalog::Schema;
+use risingwave_common::session_config::sink_decouple::SinkDecouple;
+use serde_json::Value;
+use tonic::async_trait;
+
+use super::catalog::desc::SinkDesc;
+use super::elasticsearch_opensearch_common::{
+    validate_config, ElasticSearchOpenSearchConfig, ElasticSearchOpenSearchFormatter,
+};
+use super::log_store::DeliveryFutureManagerAddFuture;
+use super::writer::{
+    AsyncTruncateLogSinkerOf, AsyncTruncateSinkWriter, AsyncTruncateSinkWriterExt,
+};
+use super::{DummySinkCommitCoordinator, Sink, SinkError, SinkParam, SinkWriterParam};
+use crate::sink::Result;
+
+pub const OPENSEARCH_SINK: &str = "opensearch_rust";
```

Comment:
Already made the parameters consistent， but not sure about migrating in this pr

Reply:
Calling it `_rust` doesn't look good because users shouldn't care about the implementation language. 

We should either
- Call it `opensearch_v2` (Then replace like https://github.com/risingwavelabs/risingwave/issues/16234)
- Directly replace it and deprecate old version like @wenym1 said. (Actually I also prefer this if possible)

Reply:
Discussed with yim and decided to replace the previous implementation in this pr, so rust's will be renamed `opensearch`, and java's will remain, or is it called opensearch_v1?

Reply:
If the new impl is called `opensearch`, and replaces the java impl, I think java impl will become dead code and we can directly remove it?